File: Python/Easy/860.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        # count number of 5 and 10 dollar bill
        # minimise the 10 dollar bill first cause need more 5 dollar bill for everything

        count10 = 0
        count5 = 0

        for bill in bills:
            if bill == 10:
                count10 += 1
                count5 -= 1
                if count5 < 0:
                    return False
            elif bill == 20:
                if count10 > 0:
                    count10 -= 1
                    count5 -= 1
                else:
                    count5 -= 3
                if count5 < 0:
                    return False
            else:
                count5 += 1

        return True

File: Java/Easy/860.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        // count number of 5 dollar bills and 10 dollars
        // for 20 dollar we shld use up the 10 dollar bills first since 5 dollar bills
        // can be used to pay everything
        int count5 = 0;
        int count10 = 0;

        for (int i = 0; i < bills.length; i++) {
            if (bills[i] == 10) {
                count5--;
                count10++;
                if (count5 < 0) {
                    return false;
                }
            } else if (bills[i] == 20) {
                if (count10 > 0) {
                    count10--;
                    count5--;
                } else {
                    count5 -= 3;
                }

                if (count5 < 0) {
                    return false;
                }
            } else {
                count5++;
            }
        }

        return true;
    }
}